### PR TITLE
feat(@kampus/gql): add terms query

### PR DESCRIPTION
**Pull request by cansirin**
Terms query added with pagination(after, before, first, last).

Used spec: https://relay.dev/graphql/connections.htm#sec-undefined.PageInfo

### CODE REVIEW

File: apps/gql/schema/resolvers/Query/SozlukQuery/terms.ts [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+  }
+
+  return terms;
+}
+
+type hasNextPageArgs = {
+  terms: SozlukTerm[];
+  first?: number | null;
+  before?: string | null;
+};
```
Comment (usirin) [inline apps/gql/schema/resolvers/Query/SozlukQuery/terms.ts:65]:
```suggestion
type hasNextPageArgs<T extends { id: string }> = {
  data: T[];
  first?: number | null;
  before?: string | null;
};
```

File: apps/gql/schema/resolvers/Query/SozlukQuery/terms.ts [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+
+function hasNextPage({ terms, before, first }: hasNextPageArgs) {
```
Comment (usirin) [inline apps/gql/schema/resolvers/Query/SozlukQuery/terms.ts:67]:
```suggestion
function hasNextPage<T extends { id: string }>({ data, before, first }: hasNextPageArgs<T>) {
```

File: apps/gql/package.json [hunk 3]
```diff
@@ -26,6 +27,7 @@
     "graphql-config": "4.5.0",
     "graphql-scalars": "1.21.3",
     "graphql-yoga": "3.9.1",
+    "object-hash": "^3.0.0",
```
Comment (Ketcap) [inline apps/gql/package.json:30]: Let's use exact version

File: apps/gql/tsconfig.json [hunk 4]
```diff
@@ -10,6 +10,12 @@
       "~/*": ["./*"]
     }
   },
-  "include": ["next-env.d.ts", "**/*.ts", "**/*.tsx", ".next/types/**/*.ts"],
+  "include": [
+    "next-env.d.ts",
+    "**/*.ts",
+    "**/*.tsx",
+    ".next/types/**/*.ts",
+    "../../packages/relay/pagination.ts"
```
Comment (Ketcap) [inline apps/gql/tsconfig.json:18]: @usirin do we need this ?

**Review by Ketcap — COMMENTED** [review]
In loader we have adjusted our usage for returns, let's not use filter boolean and return error if not found. 
Overall I'd only add tests on this one after rebase 🥳
```
if(!entity){
	return new Error(`Entity not found ${id}`)
}
```

File: apps/gql/tsconfig.json (hunk 4 above)
Comment (usirin) [reply apps/gql/tsconfig.json:18]: we shouldn't need to

**Review by usirin — APPROVED** [review]
💯 

Perfection 🥇

**Pull request merged**